File: 1063.cpp
#include <iostream>
#include <stack>
#include <vector>
using namespace std;

int main() {
    int n, i, j, k, flag;
    std::stack<int> pilha;

    while(cin >> n && n != 0) {
        while(!pilha.empty())
            pilha.pop();

        std::vector<char> entrada(n);
        std::vector<char> saida(n);
        std::vector<char> output(100);

        for(i = 0; i < n; i++)
            cin >> entrada[i];

        for(i = 0; i < n; i++)
            cin >> saida[i];

        i = 0; j = 0; k = 0; flag = 0;

        while(i <= n - 1) {
            pilha.push(entrada[i]);
            output[k++] = 'I';
            i++;

            while(!(pilha.empty()) && pilha.top() == saida[j]) {
                pilha.pop();
                output[k++] = 'R';
                j++;
                flag++;
            }
        }

        output[k] = '\0';

        if(flag == n)
            cout << output.data() << "\n";
        else
            cout << output.data() << " Impossible\n";
    }

    return 0;
}

File: 1063.java
import java.util.*;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        while (true) {
            int n = sc.nextInt();
            if (n == 0) break;

            Stack<Character> pilha = new Stack<>();
            char[] entrada = new char[n];
            char[] saida = new char[n];
            StringBuilder output = new StringBuilder();

            for (int i = 0; i < n; i++)
                entrada[i] = sc.next().charAt(0);

            for (int i = 0; i < n; i++)
                saida[i] = sc.next().charAt(0);

            int i = 0, j = 0, flag = 0;

            while (i <= n - 1) {
                pilha.push(entrada[i]);
                output.append('I');
                i++;

                while (!pilha.isEmpty() && j < n && pilha.peek() == saida[j]) {
                    pilha.pop();
                    output.append('R');
                    j++;
                    flag++;
                }
            }

            if (flag == n)
                System.out.println(output.toString());
            else
                System.out.println(output.toString() + " Impossible");
        }
        sc.close();
    }
}
